### ISSUE TITLE
Docker builds are failing due to recursion error on pd

### ISSUE THREAD

**Issue by conorsch** [penumbra-zone/penumbra#2218]
See [example failing build](https://github.com/penumbra-zone/penumbra/actions/runs/4494704940/jobs/7907512701):

```
#24 213.5    Compiling penumbra-transaction v0.1.0 (/usr/src/transaction)
#24 ...

#49 [linux/amd64->arm64 build-env 7/9] RUN if [ "arm64" = "arm64" ] && [ "amd64" != "arm64" ]; then       export CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER=aarch64-linux-gnu-gcc         CC_aarch64_unknown_linux_gnu=aarch64-linux-gnu-gcc         CXX_aarch64_unknown_linux_gnu=aarch64-linux-gnu-g++         PKG_CONFIG_SYSROOT_DIR=/usr/aarch64-linux-gnu;       cargo build --release --target aarch64-unknown-linux-gnu;     elif [ "arm64" = "amd64" ] && [ "amd64" != "amd64" ]; then       export CARGO_TARGET_x86_64_UNKNOWN_LINUX_GNU_LINKER=x86_64-linux-gnu-gcc         CC_x86_64_unknown_linux_gnu=x86_64-linux-gnu-gcc         CXX_x86_64_unknown_linux_gnu=x86_64-linux-gnu-g++         PKG_CONFIG_SYSROOT_DIR=/usr/x86_64-linux-gnu;       cargo build --release --target x86_64-unknown-linux-gnu;     else       cargo build --release --target $(uname -m)-unknown-linux-gnu;    fi;
#49 191.2    Compiling sqlx v0.6.2
#49 ...

#24 [linux/amd64 build-env 7/9] RUN if [ "amd64" = "arm64" ] && [ "amd64" != "arm64" ]; then       export CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER=aarch64-linux-gnu-gcc         CC_aarch64_unknown_linux_gnu=aarch64-linux-gnu-gcc         CXX_aarch64_unknown_linux_gnu=aarch64-linux-gnu-g++         PKG_CONFIG_SYSROOT_DIR=/usr/aarch64-linux-gnu;       cargo build --release --target aarch64-unknown-linux-gnu;     elif [ "amd64" = "amd64" ] && [ "amd64" != "amd64" ]; then       export CARGO_TARGET_x86_64_UNKNOWN_LINUX_GNU_LINKER=x86_64-linux-gnu-gcc         CC_x86_64_unknown_linux_gnu=x86_64-linux-gnu-gcc         CXX_x86_64_unknown_linux_gnu=x86_64-linux-gnu-g++         PKG_CONFIG_SYSROOT_DIR=/usr/x86_64-linux-gnu;       cargo build --release --target x86_64-unknown-linux-gnu;     else       cargo build --release --target $(uname -m)-unknown-linux-gnu;    fi;
#24 217.3    Compiling penumbra-component v0.1.0 (/usr/src/component)
#24 220.4 warning: unused import: `DomainType`
#24 220.4   --> component/src/dex/position_manager.rs:10:22
#24 220.4    |
#24 220.4 10 | use penumbra_proto::{DomainType, StateReadProto, StateWriteProto};
#24 220.4    |                      ^^^^^^^^^^
#24 220.4    |
#24 220.4    = note: `#[warn(unused_imports)]` on by default
#24 220.4 
#24 243.4    Compiling penumbra-wallet v0.1.0 (/usr/src/wallet)
#24 246.5 error[E0275]: overflow evaluating the requirement `penumbra_tct::internal::complete::node::Node<penumbra_tct::internal::complete::leaf::Leaf<penumbra_tct::internal::complete::item::Item>>: std::marker::Sync`
#24 246.5   --> pd/src/consensus/service.rs:28:14
#24 246.5    |
#24 246.5 28 |             .spawn(Worker::new(storage, queue_rx).await?.run())
#24 246.5    |              ^^^^^
#24 246.5    |
#24 246.5    = help: consider increasing the recursion limit by adding a `#![recursion_limit = "256"]` attribute to your crate (`pd`)
#24 246.5    = note: required because it appears within the type `penumbra_tct::internal::complete::node::children::shape::___C<penumbra_tct::internal::complete::node::Node<penumbra_tct::internal::complete::leaf::Leaf<penumbra_tct::internal::complete::item::Item>>>`
#24 246.5    = note: required for `Arc<penumbra_tct::internal::complete::node::children::shape::___C<penumbra_tct::internal::complete::node::Node<penumbra_tct::internal::complete::leaf::Leaf<penumbra_tct::internal::complete::item::Item>>>>` to implement `std::marker::Sync`
#24 246.5    = note: required because it appears within the type `penumbra_tct::internal::complete::node::children::Children<penumbra_tct::internal::complete::node::Node<penumbra_tct::internal::complete::leaf::Leaf<penumbra_tct::internal::complete::item::Item>>>`
#24 246.5 note: required by a bound in `tokio::task::Builder::<'a>::spawn`
#24 246.5   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.26.0/src/task/builder.rs:88:23
#24 246.5    |
#24 246.5 88 |         Fut: Future + Send + 'static,
#24 246.5    |                       ^^^^ required by this bound in `tokio::task::Builder::<'a>::spawn`
#24 246.5 
#24 246.5 For more information about this error, try `rustc --explain E0275`.
#24 246.6 error: could not compile `pd` due to previous error
#24 246.6 warning: build failed, waiting for other jobs to finish...
```

**Comment by conorsch** [comment]
Error is reproducible locally with minimal example:

```
/usr/bin/docker buildx build \
    --file deployments/containerfiles/Dockerfile \
    --platform linux/amd64,linux/arm64 \
    --tag ghcr.io/penumbra-zone/penumbra:main \
    .
```

**Comment by conorsch** [comment]
Nice, and cherry-picking the recursion fix resolves the buildx error. Pushing to main, will monitor results on the deploy.

**Comment by conorsch** [comment]
That fixed it; builds are working again: https://github.com/penumbra-zone/penumbra/actions/runs/4495572634

**Comment by conorsch** [comment]
Happened again. Looks like it was reintroduced by merge of https://github.com/penumbra-zone/penumbra/pull/2250. Investigating.

**Comment by conorsch** [comment]
Resolved in #2257.
